packages/core/src/storage.ts

```diff
@@ -20,24 +21,42 @@ export interface IClientStorage {
  */
 export class ClientStorage implements IClientStorage {
   private readonly storage: Storage;
+  private readonly prefix: string = '';
 
   constructor(storage?: Storage) {
-    this.storage = storage ?? (typeof localStorage !== 'undefined' ? localStorage : new MemoryStorage());
+    this.storage =
+      storage ?? (typeof globalThis.localStorage !== 'undefined' ? globalThis.localStorage : new MemoryStorage());
+    this.prefix = this.storage === globalThis.localStorage ? '@medplum:' : '';
```

Comment:
We should probably preserve default behavior in `@medplum/core`.

We can change default behavior in `app`, as that doesn't have any backward compatibility requirements, but `core` by itself might have people depending on the current behavior.